src/main/java/ConsentStringParser.java

```diff
@@ -0,0 +1,450 @@
+	private boolean defaultConsent;
+
+	private static Decoder decoder = Base64.getUrlDecoder();
+
+	/**
+	 * Constructor.
+	 *
+	 * @param consentString
+	 *            (required). The binary user consent data encoded as url and filename safe base64 string
+	 *
+	 * @throws ParseException
+	 *             if the consent string cannot be parsed
+	 */
+	public ConsentStringParser(String consentString) throws ParseException {
+		this(decoder.decode(consentString));
+		this.consentString = consentString;
+	}
+
+	/**
+	 * Constructor
+	 *
+	 * @param bytes:
+	 *            the byte string encoding the user consent data
+	 * @throws ParseException
+	 *             when the consent string cannot be parsed
+	 */
+	public ConsentStringParser(byte[] bytes) throws ParseException {
+		this.bits = new Bits(bytes);
+		// begin parsing
+
+		this.version = bits.getInt(VERSION_BIT_OFFSET, VERSION_BIT_SIZE);
+		String s = bits.getBinaryString();
+		this.cookieCreated = bits.getInstantFromEpochDemiseconds(CREATED_BIT_OFFSET, CREATED_BIT_SIZE);
+		this.cookieLastUpdated = bits.getInstantFromEpochDemiseconds(UPDATED_BIT_OFFSET, UPDATED_BIT_SIZE);
+		this.cmpID = bits.getInt(CMP_ID_OFFSET, CMP_ID_SIZE);
+		this.cmpVersion = bits.getInt(CMP_VERSION_OFFSET, CMP_VERSION_SIZE);
+		this.consentScreenID = bits.getInt(CONSENT_SCREEN_SIZE_OFFSET, CONSENT_SCREEN_SIZE);
+		this.consentLanguage = bits.getSixBitString(CONSENT_LANGUAGE_OFFSET, CONSENT_LANGUAGE_SIZE);
+		this.vendorListVersion = bits.getInt(VENDOR_LIST_VERSION_OFFSET, VENDOR_LIST_VERSION_SIZE);
+		this.maxVendorSize = bits.getInt(MAX_VENDOR_ID_OFFSET, MAX_VENDOR_ID_SIZE);
+		this.vendorEncodingType = bits.getInt(ENCODING_TYPE_OFFSET, ENCODING_TYPE_SIZE);
+		for (int i = PURPOSES_OFFSET, ii = PURPOSES_OFFSET + PURPOSES_SIZE; i < ii; i++) {
+			allowedPurposes.add(bits.getBit(i));
+		}
+		if (vendorEncodingType == VENDOR_ENCODING_RANGE) {
+			this.rangeEntries = new ArrayList<RangeEntry>();
+			this.defaultConsent = bits.getBit(DEFAULT_CONSENT_OFFSET);
+			int numEntries = bits.getInt(NUM_ENTRIES_OFFSET, NUM_ENTRIES_SIZE);
+			int currentOffset = RANGE_ENTRY_OFFSET;
+			for (int i = 0; i < numEntries; i++) {
+				boolean range = bits.getBit(currentOffset);
+				currentOffset++;
+				if (range) {
+					int startVendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					int endVendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					rangeEntries.add(new RangeEntry(startVendorId, endVendorId));
+				} else {
+					int vendorId = bits.getInt(currentOffset, VENDOR_ID_SIZE);
+					currentOffset += VENDOR_ID_SIZE;
+					rangeEntries.add(new RangeEntry(vendorId));
+				}
+			}
+		}
+
+	}
+
+	/**
+	 * @return the string passes in the constructor.
+	 *
+	 */
+	public String getConsentString() {
+		return consentString;
+	}
+
+	/**
+	 * @return the {@link Instant} at which the cookie was created
+	 */
+	public Instant getCookieCreated() {
```

Comment:
The IAB has moved away from the use of the word "Cookie" to refer to the Consent String because that describes the mechanism by which the value is stored.